src/integrations/buckaroo/class-buckaroo.php

```diff
@@ -0,0 +1,571 @@
+      throw Missing_Filter_Info_Exception::view_exception( Filters::BUCKAROO, 'website_key' );
+    }
+
+    if ( empty( \apply_filters( Filters::BUCKAROO, 'secret_key' ) ) ) {
+      throw Missing_Filter_Info_Exception::view_exception( Filters::BUCKAROO, 'secret_key' );
+    }
+  }
+
+  /**
+   * Get the correct url depending on if we're testing or not.
+   *
+   * @return string
+   */
+  protected function get_buckaroo_uri(): string {
+    return $this->is_test() ? $this->get_buckaroo_uri_test() : $this->get_buckaroo_uri_live();
+  }
+
+  /**
+   * Returns correct Buckaroo live uri.
+   *
+   * @return string
+   */
+  protected function get_buckaroo_uri_live(): string {
+    return $this->is_data_request() ? self::LIVE_URI_DATA_REQUEST : self::LIVE_URI_TRANSACTION;
+  }
+
+  /**
+   * Returns correct Buckaroo test uri.
+   *
+   * @return string
+   */
+  protected function get_buckaroo_uri_test(): string {
+    return $this->is_data_request() ? self::TEST_URI_DATA_REQUEST : self::TEST_URI_TRANSACTION;
+  }
+
+  /**
+   * Check if we're running a test or not.
+   *
+   * @return boolean
+   */
+  protected function is_test(): bool {
+    return $this->is_test_uri;
+  }
+
+  /**
+   * Check if we're running a test or not.
+   *
+   * @return boolean
+   */
+  protected function is_data_request(): bool {
+    return $this->is_data_request;
+  }
+
+  /**
+   * Set if you need to use the test URI instead of live one.
+   *
+   * @return void
+   */
+  public function set_test(): void {
+    $this->is_test_uri = true;
+  }
+
+  /**
+   * Set if you need to use the test URI instead of live one.
+   *
+   * @return void
+   */
+  public function set_data_request(): void {
+    $this->is_data_request = true;
+  }
+
+  /**
+   * Set's currency as uppercase 3-letter string (example: EUR)
+   *
+   * @param  string $currency Currency string.
+   * @return void
+   */
+  public function set_currency( string $currency ): void {
+    $this->currency = $currency;
+  }
+
+  /**
+   * Getter for $this->currency.
+   *
+   * @return string
+   */
+  public function get_currency(): string {
+    return $this->currency;
+  }
+
+  /**
+   * Get type of payment. Defaults to iDEAL.
+   *
+   * @return  string
+   */
+  public function get_pay_type() {
+     return $this->pay_type;
+  }
+
+  /**
+   * Set type of payment. Defaults to iDEAL.
+   *
+   * @param  string $pay_type  Type of payment. Defaults to iDEAL.
+   *
+   * @return  self
+   */
+  public function set_pay_type( string $pay_type ) {
+    $this->pay_type = $pay_type;
+    return $this;
+  }
+
+  /**
+   * Get return URL after payment.
+   *
+   * @return  string
+   */
+  public function get_return_url() {
+    return $this->return_url;
+  }
+
+  /**
+   * Set return URL after payment.
+   *
+   * @param  string $return_url  Return URL after payment.
+   *
+   * @return  self
+   */
+  public function set_return_url( string $return_url ) {
+    $this->return_url = $return_url;
+
+    return $this;
```

Comment:
Same here. Is this used for chaining purposes?

Reply:
See comment above ⬆️